source/Calamari.Tests/KubernetesFixtures/InstallTools.cs

```diff
@@ -217,6 +218,45 @@ public async Task InstallGCloud()
                         return GetGcloudExecutablePath(destinationDirectoryName);
                     });
             }
+            
+            if (CalamariEnvironment.IsRunningOnWindows)
+            {
+                InstallGkeAuthPlugin();
+            }
+        }
+        
+        void InstallGkeAuthPlugin()
+        {
+            var variables = new Dictionary<string, string>();
+            var pythonCopyPath = ExecuteCommandAndReturnResult(GcloudExecutable, "components copy-bundled-python", ".", variables);
+            variables.Add("CLOUDSDK_PYTHON", pythonCopyPath.commandOutput);
+
+            var gkeComponent = ExecuteCommandAndReturnResult($"\"{GcloudExecutable}\"", "components list --filter=\"Name=gke-gcloud-auth-plugin\" --format=\"json\"", ".", variables);
+            var gkeComponentJObject = JArray.Parse(gkeComponent.commandOutput).First();
+            var installedState = gkeComponentJObject["state"]["name"].Value<string>();
+            if (installedState != "Installed")
+            {
+                ExecuteCommandAndReturnResult(GcloudExecutable, "components install gke-gcloud-auth-plugin --quiet", ".", variables);
+            }
+        }
+
+        (int ExitCode, string commandOutput) ExecuteCommandAndReturnResult(string executable, string arguments, string workingDirectory, Dictionary<string, string> variables)
+        {
+            var stdOut = new StringBuilder();
+            var stdError = new StringBuilder();
+            var commandExitCode = SilentProcessRunner.ExecuteCommand(executable,
+                                                                     arguments,
+                                                                     workingDirectory,
+                                                                     variables,
+                                                                     (Action<string>)(s => stdOut.AppendLine(s)),
+                                                                     (Action<string>)(s => stdError.AppendLine(s))).ExitCode;
+
+            if (commandExitCode != 0)
```

Comment:
If whenever the exit code is not 0 an exception is thrown, that means the exit code in return value will always be 0. So we don't really need the `ExitCode` field in the result, and we are not using it anyway.

Reply:
@YihaoOct I'm not sure exactly what you are referring to here. As you saying the `SilentProcessRunner.ExecuteCommand` will throw an exception on a non-zero exit code? I couldn't see that in the code

Reply:
This PR has been a bit old 😅 In the original version I commented on: https://github.com/OctopusDeploy/Calamari/pull/971/commits/5fb98b5a02daa6cb6fd0d67561d595e99aaa8970

The ExitCode used to be part of the return value of this method. I was saying that if we can reach the return statement without an exception, then it means the ExitCode has to be 0. Therefore we don't need to return it.

It's not a concern now, but because of the place I put the comment on, it wan't picked up as `Outdated`

Reply:
ah, all good 👍